src/main/java/com/user_service/controllers/UserAuthenticationController.java

```diff
@@ -16,13 +16,13 @@ public class UserAuthenticationController {
     private UserAuthenticationService userAuthenticationService;
 
     @PostMapping(value = "/register")
-    public ResponseEntity<String> signUpUser(@RequestBody RegisterUserDto registerUserDto) {
-        SignUpResponseDto signUpResponseDto = userAuthenticationService.signUpUser(registerUserDto);
+    public ResponseEntity<String> signUpUser(@RequestBody SignUpUserDto signUpUserDto) {
+        SignUpResponseDto signUpResponseDto = userAuthenticationService.signUpUser(signUpUserDto);
 
         if (signUpResponseDto.getSignUpResponse() == null) {
-            return new ResponseEntity<String>(signUpResponseDto.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+            return new ResponseEntity<>(signUpResponseDto.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
         }
-        return new ResponseEntity<String>("Account sign up, confirmation code sent to " + registerUserDto.getEmail(), HttpStatus.OK);
+        return new ResponseEntity<>("Account sign up, confirmation code sent to " + signUpUserDto.getEmail(), HttpStatus.OK);
```

Comment:
I think you should return a JSON response with things like `user_id` or sth similar. Plain strings are not very useful in REST API and they lack structure.